src/app.js

```diff
@@ -6,8 +6,236 @@ import _ from 'underscore';
 import './css/foundation.css';
 import './css/style.css';
 
+import TripList from './app/collections/trip_list';
+import Trip from './app/models/trip';
+import Reservation from './app/models/reservation';
+
 console.log('it loaded!');
 
+const tripList = new TripList()
+// const trip = new Trip()
+// const res = new Reservation()
+
+const TRIP_FIELDS = ['name', 'category', 'continent', 'weeks', 'cost', 'about'];
+const RES_FIELDS = ['name', 'email', 'tripId'];
+
+
+let tripsTemplate;
+let tripDetailTemplate;
+
+// Clear status messages
+const clearStatus = function clearStatus() {
+  $('#status-messages ul').html('');
+  $('#status-messages').hide();
+};
+
+// Add a new status message
+const reportStatus = function reportStatus(status, message) {
+  console.log(`Reporting ${ status } status: ${ message }`);
+
+  // Should probably use an Underscore template here.
+  const statusHTML = `<li class="${ status }">${ message }</li>`;
+
+  // note the symetry with clearStatus()
+  $('#status-messages ul').append(statusHTML);
+  $('#status-messages').show();
+};
+
+const render = function render(tripList) {
+  const $tripsElement = $('#trip-list');
+  $tripsElement.html('');
+
+  let tripId = $(this).attr('data-id');
+  let trip = tripList.get(tripId)
+
+  tripList.forEach((trip) => {
+    const generatedHTML = $(tripsTemplate(trip.attributes));
+    generatedHTML.on('click', (event) => {
+      console.log('testing click trip detail');
+      $('#trips').hide()
+      $('#trip').show()
+      // $('#add-res').show()
+      // due to asynchonous api responses we put the renderTrip into the .fetch()
+      trip.fetch({
+        success(model, response) {
+          renderTrip(model);
+        }
+      });
+
+    })
+    $tripsElement.append(generatedHTML)
+  });
+};
+
+
+const renderTrip = function renderTrip(trip) {
+  const $tripElement = $('#trip');
+  $tripElement.html('');
+  console.log(trip.attributes);
+  // console.log(trip.attributes.about);
+
+  const generatedDetailHTML = tripDetailTemplate(trip.attributes);
+  $tripElement.html(generatedDetailHTML)
+
+  $('#add-res').on('submit', addResHandler);
+
+  // Provide visual feedback for sorting
+  $('th.sort').removeClass('current-sort-field');
+  $(`th.sort.${ tripList.comparator }`).addClass('current-sort-field');
+}
+
+const loadTrips = function loadTrips() {
+  tripList.fetch({
+    success(model, response) {
+    render(model);
+    }
+  });
+};
+
+const readTripFormData = function readTripFormData() {
+  const tripData = {};
+  TRIP_FIELDS.forEach((field) => {
+    // select the input corresponding to the field we want
+    const $inputElement = $(`#add-trip-form input[name="${ field }"]`);
+    const value = $inputElement.val();
+
+    // Don't take empty strings, so that Backbone can
+    // fill in default values
+    if (value != '') {
+      tripData[field] = value;
+    }
+
+    $inputElement.val('');
```

Comment:
Consider this scenario:

1).  I click on a trip, great the form shows up.
2).  forget to put in my email address before hitting submit, I get a validation error message, great
3).  I notice the fields have their values blanked, also ok as I can reenter them (just annoying), but the hidden tripId field is also blanked now the submission fails.  

A better time to do this is after a successful save, just make a helper to blank the form and call it after a successful save.